examples/animation/animation_graph.rs

```diff
@@ -0,0 +1,538 @@
+    /// The width of this rectangle in pixels.
+    width: f32,
+    /// The height of this rectangle in pixels.
+    height: f32,
+}
+
+/// Either a straight horizontal or a straight vertical line on screen.
+///
+/// The line starts at (`left`, `bottom`) and goes either right (if the line is
+/// horizontal) or down (if the line is vertical).
+struct Line {
+    /// The number of pixels that the start of this line is from the left edge
+    /// of the screen.
+    left: f32,
+    /// The number of pixels that the start of this line is from the bottom edge
+    /// of the screen.
+    bottom: f32,
+    /// The length of the line.
+    length: f32,
+}
+
+/// Assets that the example needs to use.
+#[derive(Resource)]
+struct AppAssets {
+    /// The [`AnimationGraph`] asset, which specifies how the animations are to
+    /// be blended together.
+    animation_graph: Handle<AnimationGraph>,
+    /// Indices of the three animations in the graph.
+    ///
+    /// This value is populated when the graph is loaded.
+    node_indices: [AnimationNodeIndex; 3],
+}
+
+/// The current weights of the three playing animations.
+#[derive(Resource)]
+struct AnimationWeights {
+    /// The weights of the three playing animations.
+    weights: [f32; 3],
+}
+
+/// The type of each node in the UI: either a clip node or a blend node.
+enum NodeType {
+    /// A clip node, which specifies an animation.
+    Clip(ClipNodeText),
+    /// A blend node with no animation and a string label.
+    Blend(&'static str),
+}
+
+/// The label for the UI representation of a clip node.
+#[derive(Clone, Component)]
+struct ClipNodeText {
+    /// The string label of the node.
+    text: &'static str,
+    /// Which of the three animations this UI widget represents.
+    index: usize,
+}
+
+/// Whether, and where, the mouse is dragging.
+#[derive(Resource, Default)]
+enum DragState {
+    /// The mouse isn't being dragged.
+    #[default]
+    NotDragging,
+    /// The mouse is being dragged.
+    Dragging {
+        /// Which animation the mouse is dragging.
+        clip_index: usize,
+        /// The X position of the mouse when the drag started.
+        reference_x_pos: f32,
+        /// The weight that the dragged node had when the drag started.
+        reference_weight: f32,
+    },
+}
+
+/// How much weight a drag of one mouse pixel corresponds to.
+const WEIGHT_ADJUST_SPEED: f32 = 0.01;
+
+/// The help text in the upper left corner.
+static HELP_TEXT: &str = "Click and drag an animation clip node to change its weight";
+
+/// The node widgets in the UI.
+static NODE_TYPES: [NodeType; 5] = [
+    NodeType::Clip(ClipNodeText::new("Idle", 0)),
+    NodeType::Clip(ClipNodeText::new("Walk", 1)),
+    NodeType::Blend("Root"),
+    NodeType::Blend("Blend\n0.5"),
+    NodeType::Clip(ClipNodeText::new("Run", 2)),
+];
+
+/// The positions of the node widgets in the UI.
+///
+/// These are in the same order as [`NODE_TYPES`] above.
+static NODE_RECTS: [NodeRect; 5] = [
```

Comment:
Why are we using `static` rather than `const` in this example?

Reply:
I usually prefer `static` for big things as `const` has a tendency to make more temporary copies onto the stack.